base/abstractarray.jl

```diff
@@ -1868,11 +1868,13 @@ julia> mapslices(sum, a, [1,2])
  58
 ```
 """
-mapslices(f, A::AbstractArray, dims) = mapslices(f, A, [dims...])
-function mapslices(f, A::AbstractArray, dims::AbstractVector)
+function mapslices(f, A::AbstractArray; dims)
     if isempty(dims)
         return map(f,A)
     end
+    if !isa(dims, AbstractVector)
+        dims = [dims...]
```

Comment:
This introduces a new instability here, but this is _already_ an unstable "outer" function with a stable kernel, so I was hoping it wouldn't affect things too much.  Unfortunately it seems to have quite a large effect (6-10x) in my cursory tests:

```julia
# Commit 948b088f17 (3 days old master)
julia> g(f, A, dims) =  mapslices(f, A, dims)
g (generic function with 1 method)

julia> A = rand(500,500);

julia> @benchmark g(sum, $A, 1)
BenchmarkTools.Trial:
  memory estimate:  111.84 KiB
  allocs estimate:  4040
  --------------
  minimum time:     713.670 μs (0.00% GC)
  median time:      729.114 μs (0.00% GC)
  mean time:        750.090 μs (2.53% GC)
  maximum time:     51.052 ms (98.37% GC)
  --------------
  samples:          6637
  evals/sample:     1

# jb/mapslicesdims/d184a46949 (fork: 1 commits, 0 days)
julia> g(f, A, dims) =  mapslices(f, A, dims=dims)
g (generic function with 1 method)

julia> A = rand(500,500);

julia> @benchmark g(sum, $A, 1)
BenchmarkTools.Trial:
  memory estimate:  345.75 KiB
  allocs estimate:  15517
  --------------
  minimum time:     4.575 ms (0.00% GC)
  median time:      4.606 ms (0.00% GC)
  mean time:        4.718 ms (1.90% GC)
  maximum time:     55.249 ms (91.39% GC)
  --------------
  samples:          1059
  evals/sample:     1
```